x-pack/plugins/alerting/server/test_utils/index.ts

```diff
@@ -5,20 +5,18 @@
  */
 
 interface Resolvable<T> {
-  resolve: (arg?: T) => void;
+  resolve: (arg: T) => void;
 }
 
 /**
  * Creates a promise which can be resolved externally, useful for
  * coordinating async tests.
  */
 export function resolvable<T>(): Promise<T> & Resolvable<T> {
-  let resolve: (arg?: T) => void;
-  const result = new Promise<T>(r => {
-    resolve = r;
-  }) as any;
-
-  result.resolve = (arg: T) => resolve(arg);
-
-  return result;
+  let resolve: (arg: T) => void;
+  return Object.assign(new Promise<T>(r => (resolve = r)), {
+    resolve(arg: T) {
+      return setTimeout(() => resolve(arg), 0);
+    },
+  });
```

Comment:
Removing an `any` in a test that was using `resolvable` revealed the typing was wrong here.